sound/soc/sof/pm.c

```diff
@@ -203,16 +207,22 @@ static int sof_suspend(struct device *dev, bool runtime_suspend)
 	}
 
 	target_state = snd_sof_dsp_power_target(sdev);
+	pm_state.event = target_state;
 
 	/* Skip to platform-specific suspend if DSP is entering D0 */
-	if (target_state == SOF_DSP_PM_D0)
+	if (target_state == SOF_DSP_PM_D0) {
+		sof_suspend_clients(sdev, pm_state);
```

Comment:
is this conceptually broken?

The PM frameworks will suspend children devices first, so this could be a no-op if these children devices rely on power management.

Reply:
It will either suspends the child before the core's suspend or it will never suspends it.
If the child uses pm_runtime actively then the core can not be suspended while any of the children is active (pm_runtime_get() propagates from child to parent).
dma-trace is not an active client, it will be not resumed by pm_runtime, _ever_ and it is never going to suspend either, the pm calls are ignored for it.

Reply:
ack, again something like 'suspend clients not managed by pm frameworks' would help.

Reply:
OK, I will add comments.